### PR TITLE
Mysterious Generation Machine

### PR DESCRIPTION
**Pull request by DreamwalkerSisyphe**
The "lore" is that it was a powerful machine created by some omnipotent/omniscient being with the ability to create any item typed into it. The caveat? Some prankster replaced the keyboard with a lever that generates a random string of letters instead. So instead you have to get lucky and hope it spells out a word somewhere in there. Admin spawn only for now until I work out how to properly implement it (via a hard to reach ruin, etc.)

### CODE REVIEW

**Comment by Bawhoppen** [comment]
Neat

File: code/game/objects/structures/mysterymachine.dm [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	var/itemText = ""
+	var/lastText = ""
+	var/firstText = ""
+
+/obj/structure/mysterymachine/Initialize()
+	. = ..()
+	resistance_flags |= INDESTRUCTIBLE
+	set_light(2, 2, "#FFFFFF")
+	if(!GLOB.possibleMMItems.len)
+		var/list/MMItems_types_list = subtypesof(/obj/item)
```
Comment (Iamgoofball) [inline code/game/objects/structures/mysterymachine.dm]: `var/list/MMItems_types_list = subtypesof(/obj/item)`

`subtypesof(/obj/item)`

Comment (Iamgoofball) [reply code/game/objects/structures/mysterymachine.dm]: this may have been marked as spam but it is educational and i highly advise you show comment

Comment (Iamgoofball) [reply code/game/objects/structures/mysterymachine.dm]: and i know a rookie coder will look at this down the line and think "Well its safe to use here what about elsewhere"

no it is not

it is not safe

do not do this without extreme care and caution

**Comment by octareenroon91** [comment]
Stop this and turn back.

This concept is resource intensive, and in fact wasteful, and disappointing if not boring.

For a random 32-letter input, there's maybe a 1 in 100000 chance of getting soap. Most items have names longer than 'soap'. A few have names(especially sanitized names) shorter than 'soap'. Included on this list is 'd20', 'c4', and 'x4'. Each of these have a very high chance of hitting every lever pull given 32-letter input trying to match one of 26 letters.

... and the branch is fucked now that upstream/master was pulled on top of it.

File: code/game/objects/structures/mysterymachine.dm [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			SStgui.update_uis(src)
+			. = TRUE
+	update_icon()
+
+/obj/structure/mysterymachine/proc/get_valid_items()
+	var/valid_items_to_return = new /list(0)
+	for(var/V in GLOB.possibleMMItems)
+		var/obj/item/I = V
+		var/sanitizedItemName = sanitizeName(initial(I.name))
+		if(findtext(item_input, sanitizedItemName))
```
Comment (ninjanomnom) [inline code/game/objects/structures/mysterymachine.dm]: You need to not base the entire functionality of this on finding item names in a random string. Going through every item like this is frankly horrific as I talked about in irc. Unless you change this to be more sane this isn't happening.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Pull request closed without merging**